mpp/include/mpp_util.inc

```diff
@@ -1408,6 +1408,74 @@ end function rarray_to_char
 
   end function get_ascii_file_num_lines
 
+  !#######################################################################
+  function get_ascii_file_num_lines_and_length(FILENAME, PELIST)
```

Comment:
I kept it in here because it works in conjunction with the read_ascii_file subroutine in here where one cannot be called without the other: hence the use of the read_ascii_file_on logical variable. It's sort of a mess, but you need to read an ASCII file twice to properly absorb its contents with Fortran.

I'll create Doxygen for this and the other routines that I've copied and pasted out of fms_io.

Reply:
The other alternative is to move read_ascii_file over to fms2_io (along with get_ascii_file_num_lines_and_length). Let me know if you have a preference.

Reply:
I think it could be combined (I guess the old function would become obsolete?), I just wasn't sure if we wanted to make that call for read_input_nml?

Reply:
Even if we don't rework the input_nml_file variable, we can use this new function to ensure get the two pieces of info (num lines & line length).  From there we could check to make sure the parameter is "wide enough" to hold the individual lines.  The use of num lines is obvious.

Reply:
I just pushed a new commit that uses the more aggressive, integrated approach (essentially deprecating INPUT_STR_LENGTH), and it worked with the fms2_io unit tests.

Reply:
@GFDL-Eric Nice update.  I think we can look to see if the get_ascii_file_num_lines function is used anywhere and delete it in favor of this more comprehensive version.

Reply:
The only place I see it in FMS is in diag_table.F90, but that might be addressed with one of the recent PRs?